pkg/appvalidation/business_case.go

```diff
@@ -76,3 +76,143 @@ func BusinessCaseForUpdate(businessCase *models.BusinessCase) error {
+		}
+		if validate.RequireNullString(businessCase.PreferredSummary) {
+			expectedErr.WithValidation("PreferredSummary", "is required")
+		}
+		if validate.RequireNullString(businessCase.PreferredAcquisitionApproach) {
+			expectedErr.WithValidation("PreferredAcquisitionApproach", "is required")
+		}
+		if validate.RequireNullString(businessCase.PreferredPros) {
+			expectedErr.WithValidation("PreferredPros", "is required")
+		}
+		if validate.RequireNullString(businessCase.PreferredCons) {
+			expectedErr.WithValidation("PreferredCons", "is required")
+		}
+		if validate.RequireNullString(businessCase.PreferredCostSavings) {
+			expectedErr.WithValidation("PreferredCostSavings", "is required")
+		}
+		if validate.RequireNullString(businessCase.AlternativeATitle) {
+			expectedErr.WithValidation("AlternativeATitle", "is required")
+		}
+		if validate.RequireNullString(businessCase.AlternativeASummary) {
+			expectedErr.WithValidation("AlternativeASummary", "is required")
+		}
+		if validate.RequireNullString(businessCase.AlternativeAAcquisitionApproach) {
+			expectedErr.WithValidation("AlternativeAAcquisitionApproach", "is required")
+		}
+		if validate.RequireNullString(businessCase.AlternativeAPros) {
+			expectedErr.WithValidation("AlternativeAPros", "is required")
+		}
+		if validate.RequireNullString(businessCase.AlternativeACons) {
+			expectedErr.WithValidation("AlternativeACons", "is required")
+		}
+		if validate.RequireNullString(businessCase.AlternativeACostSavings) {
+			expectedErr.WithValidation("AlternativeACostSavings", "is required")
+		}
+		if businessCase.SubmittedAt == nil {
+			expectedErr.WithValidation("SubmittedAt", "is required")
+		} else if validate.RequireTime(*businessCase.SubmittedAt) {
+			expectedErr.WithValidation("SubmittedAt", "is cannot be zero")
+		}
+		if businessCase.AlternativeBTitle.Valid ||
+			businessCase.AlternativeBSummary.Valid ||
+			businessCase.AlternativeBAcquisitionApproach.Valid ||
+			businessCase.AlternativeBPros.Valid ||
+			businessCase.AlternativeBCons.Valid ||
+			businessCase.AlternativeBCostSavings.Valid {
+			if validate.RequireNullString(businessCase.AlternativeBTitle) {
+				expectedErr.WithValidation("AlternativeBTitle", "is required")
+			}
+			if validate.RequireNullString(businessCase.AlternativeBSummary) {
+				expectedErr.WithValidation("AlternativeBSummary", "is required")
+			}
+			if validate.RequireNullString(businessCase.AlternativeBAcquisitionApproach) {
+				expectedErr.WithValidation("AlternativeBAcquisitionApproach", "is required")
+			}
+			if validate.RequireNullString(businessCase.AlternativeBPros) {
+				expectedErr.WithValidation("AlternativeBPros", "is required")
+			}
+			if validate.RequireNullString(businessCase.AlternativeBCons) {
+				expectedErr.WithValidation("AlternativeBCons", "is required")
+			}
+			if validate.RequireNullString(businessCase.AlternativeBCostSavings) {
+				expectedErr.WithValidation("AlternativeBCostSavings", "is required")
+			}
```

Comment:
I think we need some check that if Alternative B is being used, then the lifecycle cost lines associated are present. And similarly, if Alternative B is not being used, the lifecycle cost items are not present.

Reply:
I think I've added some of that. I didn't do the similarly one, though